sqlite3/lib/src/wasm/js_interop.dart

```diff
@@ -34,6 +34,24 @@ extension JsContext on _JsContext {
   external IdbFactory? get indexedDB;
 }
 
+extension IdbFactoryExt on IdbFactory {
+  Future<List<DatabaseName>> databases() async {
+    final jsDatabases = await promiseToFutureAsMap(
+        callMethod<Object>(this, 'databases', const []));
+    return jsDatabases!.values.whereType<Map<String, dynamic>>().map((jsMap) {
+      final value = jsMap.values.toList();
+      return DatabaseName(value[0] as String, value[1] as int);
+    }).toList();
+  }
+}
+
+class DatabaseName {
+  final String name;
+  final int version;
+
+  DatabaseName(this.name, this.version);
+}
```

Comment:
Would something like this work?

```suggestion
@JS()
@anonymous
class DatabaseName {
  external String get name;
  external int get version;
}
```

When we can just do `promiseToFuture<DatabaseName>()` and don't have to go over the map?

Reply:
I'm new in JS interop. I tried a solution similar to this, but didn't worked. I try again